src/gen9_avc_encoder.c

```diff
@@ -860,7 +992,7 @@ gen9_avc_allocate_resources(VADriverContextP ctx,
         goto failed_allocation;
 
     /* scaling related surface   */
-    if (avc_state->mb_status_supported) {
+    if (!IS_GEN8(i965->intel.device_info)) {
```

Comment:
it looks confused and "if (avc_state->mb_status_supported)"  looks better. would you initialize the  variable mb_status_supported according to the GEN8 platform